provider/aws/aws.go

```diff
@@ -494,9 +497,30 @@ func (p *AWSProvider) submitChanges(ctx context.Context, changes []*route53.Chan
 	for z, cs := range changesByZone {
 		var failedUpdate bool
 
-		batchCs := batchChangeSet(cs, p.batchChangeSize)
+		// group changes into new changes and into changes that failed in a previous iteration and are retried
+		var newChanges, retriedChanges []*route53.Change
+		for _, c := range cs {
+			found := false
+			if p.failedChangesQueue[z] != nil {
```

Comment:
I think this check is useless, if `p.failedChangesQueue[z]` the range will never run. Please remove it.

Reply:
The range simply goes over _all_ desired changes (no matter if they have failed before - because the loop is finding out _if_ they have failed before and sort them into new changes / changes that have failed before). Therefor, `p.failedChangesQueue[z]` is actually most likely to be `nil`.

Reply:
I understand, but that's no problem for the range that you are doing the line below, see:

```
package main

import (
	"fmt"
)

func main() {
	var nilMap map[string]string
	nilMap = nil // making this more explicit that it needs to be :-)
	for _, val := range nilMap{
		fmt.Println(val)
	}
}
```

Reply:
I see, didn't know that `for` works on a `nil` map :upside_down_face: Will remove the check.

Reply:
@alfredkrohmer as far as I see this was not done.

Reply:
@alfredkrohmer I agree, let's remove the `if`. Should be fine.
@szuecs There should be no need for a lock since it all runs sequentially.